generator/sbpg/targets/resources/SbpTemplate.hs

```diff
@@ -121,25 +122,23 @@ instance Binary SBPMsg where
       encode' (SBPMsgBadCrc  sbp) = sbp
 
 instance FromJSON SBPMsg where
-  parseJSON (Object o) = case lookup "msg_type" o of
-    Nothing         -> mzero
-    Just (Number msgType) -> decode' (toBoundedInteger msgType) where
-                               decode' Nothing = mzero
-                               decode' (Just t)
-                         ((*- for m in msgs *))
-                                 | t == (((m | to_global))) = SBP(((m))) <$> parseJSON (Object o) <*> parseJSON (Object o)
-                         ((*- endfor *))
-                                 | otherwise = SBPMsgUnknown <$> parseJSON (Object o)
-    Just _ -> mzero
-  parseJSON (_) = mzero
+  parseJSON obj@(Object o) = do
+    msgType <- o .: "msg_type"
+    decode' msgType where
+      decode' msgType
+((*- for m in msgs *))
+        | msgType == (((m | to_global))) = SBP(((m))) <$> parseJSON obj <*> parseJSON obj
+((*- endfor *))
+        | otherwise = SBPMsgUnknown <$> parseJSON obj
+  parseJSON _ = mzero
 
 merge :: Value -> Value -> Value
-merge (Object one) (Object two) = Object $ one <> two
-merge _ _ = undefined
+merge (Object one) (Object two) = Object (one <> two)
+merge _ _ = Null
```

Comment:
Hmm, yeah - was wondering about that.... maybe something like this???

```haskell
merge :: Value -> Value -> Value
merge (Object one) (Object two) = Object (one <> two)
merge obj@(Object _) _ = obj
merge _ obj@(Object _) = obj
merge obj Null = obj
merge _ obj = obj
```

Definitely think we should yank the `undefined` either way.